devicetypes/smartthings/zwave-door-temp-sensor.src/zwave-door-temp-sensor.groovy

```diff
@@ -0,0 +1,299 @@
+			input "tempOffset", "number", title: "Degrees", description: "Adjust temperature by this many degrees", range: "*..*", displayDuringSetup: false
+		}
+	}
+*/
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"contact", type: "generic", width: 6, height: 4){
+			tileAttribute ("device.contact", key: "PRIMARY_CONTROL") {
+				attributeState("open", label: '${name}', icon: "st.contact.contact.open", backgroundColor: "#e86d13")
+				attributeState("closed", label: '${name}', icon: "st.contact.contact.closed", backgroundColor: "#00A0DC")
+			}
+			tileAttribute("device.temperature", key: "SECONDARY_CONTROL") {
+				attributeState("default", label:'${currentValue}°',
+						backgroundColors: [
+								[value: 31, color: "#153591"],
+								[value: 44, color: "#1e9cbb"],
+								[value: 59, color: "#90d2a7"],
+								[value: 74, color: "#44b621"],
+								[value: 84, color: "#f1d801"],
+								[value: 95, color: "#d04e00"],
+								[value: 96, color: "#bc2323"]
+						]
+				)
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		main "contact"
+		details(["contact", "battery"])
+	}
+}
+
+private getCommandClassVersions() {
+	[0x20: 1, 0x25: 1, 0x30: 1, 0x31: 5, 0x80: 1, 0x84: 1, 0x71: 3, 0x9C: 1]
+}
+
+def parse(String description) {
+	def result = null
+	if (description.startsWith("Err 106")) {
+		if (state.sec) {
+			log.debug description
+		} else {
+			result = createEvent(
+				descriptionText: "This sensor failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+				eventType: "ALERT",
+				name: "secureInclusion",
+				value: "failed",
+				isStateChange: true,
+			)
+		}
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "parsed '$description' to $result"
+	return result
+}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 482min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 4 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def updated() {
+	// Device-Watch simply pings if no device events received for 482min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 4 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	def cmds = []
+	if (!state.MSR) {
+		cmds = [
+			command(zwave.manufacturerSpecificV2.manufacturerSpecificGet()),
+			"delay 1200",
+			zwave.wakeUpV1.wakeUpNoMoreInformation().format()
+		]
+	} else if (!state.lastbat) {
```

Comment:
if we don't do the manufacturer get, are we expecting the device to send us an unsolicited battery report?

Just not certain that all the possibilities end in a `NoMoreInformation`.